addons/mail/static/src/models/welcome_view/welcome_view.js

```diff
@@ -0,0 +1,127 @@
+/** @odoo-module **/
+
+import { attr } from '@mail/model/model_field';
+import { registerNewModel } from '@mail/model/model_core';
+
+function factory(dependencies) {
+
+    class WelcomeView extends dependencies['mail.model'] {
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @static
+         * @param {string} guestName 
+         * @returns {string}
+         */
+        static sanitizeGuestName(guestName = "") {
+            return guestName.trim();
+        }
+
+        /**
+         * @static
+         * @param {string} guestName 
+         */
+        static async updateGuestNameServerSide(guestName) {
+            await this.env.services.rpc({
+                route: '/mail/guest/update_name',
+                params: {
+                    guest_name: this.messaging.models['mail.welcome_view'].sanitizeGuestName(guestName),
```

Comment:
just name as param, and if you bother to make a low level rpc calling method, sanitize should be done before calling the method